client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,68 +14,66 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
 
-	getInitialState() {
-		return {
+	constructor() {
```

Comment:
We don't need a constructor for default state. We can use a class property here instead:

```jsx
static defaultProps = {
    onAddMedia: noop,
    onClose: noop,
};
state = {
    value: '',
    isError:false
};
```